lib/devise-security/validators/complexity_validator.rb

```diff
@@ -0,0 +1,28 @@
+module DeviseSecurity
+  # String complexity validator
+  # Validates complexity of e.g. passwords
+  # Options:
+  # - digit:  minimum number of digits in the validated string
+  # - lower:  minimum number of lower-case letters in the validated string
+  # - symbol: minimum number of punctuation characters or symbols in the validated string
+  # - upper:  minimum number of upper-case letters in the validated string
+  class ComplexityValidator < ActiveModel::EachValidator
```

Comment:
name this `PasswordComplexityValidator`.  You'll also need to rename all the associated tests/files.